huxley/www/js/components/ChairSummaryView.js

```diff
@@ -0,0 +1,205 @@
+      this.setState({countries: CountryStore.getCountries()});
+    });
+  },
+
+  componentWillUnmount() {
+    this._countriesToken && this._countriesToken.remove();
+    this._delegatesToken && this._delegatesToken.remove();
+    this._assignmentsToken && this._assignmentsToken.remove();
+  },
+
+  render() {
+    return (
+      <InnerView>
+        <h2>Summaries</h2>
+        <p>
+          Here you can provide feedback for the delegates. You can save any 
+          changes with the "Save" button and they will not become visible to 
+          advisors until you next publish. Please note that clicking the 
+          "Publish" button will allow advisors to read the summaries you have 
+          written for their delegates. 
+        </p>
+        <p>
+          <strong>
+            Only one chair at a time should be logged in. Changes may be lost 
+            otherwise.
+          </strong>
+        </p>
+        <form>
+          <div className="table-container">
+            <table className="table highlight-cells">
+              <thead>
+                <tr>
+                  <th>Assignment</th>
+                  <th>Summary</th>
+                </tr>
+              </thead>
+              <tbody>
+                {this.renderSummaryRows()}
+              </tbody>
+            </table>
+          </div>
+        </form>
+        <div className="foot-bar" >
+          <ul className="right">
+            <Button
+              color="green"
+              onClick={this._handleSaveSummaries}>
+              Save
+            </Button>
+            <Button
+              color="blue"
+              onClick={this._handlePublishSummaries}>
+              Publish
+            </Button>
+          </ul>
+        </div>
+      </InnerView>
+    );
+  },
+
+  renderSummaryRows() {
+    var committeeCountries = Object.keys(this.state.country_assignments);
+    var countries = this.state.countries;
+    return committeeCountries.map(country => {
+      var country_delegates = this.state.country_assignments[country];
+      return (
+        <tr key={country}>
+          <td>
+            {Object.keys(countries).length ? countries[country].name : country}
+          </td>
+          <td>
+            <textarea
+              className="text-input"
+              style={{"width": "95%"}}
+              rows="3"
+              onChange={this._handleSummaryChange.bind(this, country)}
+              defaultValue={country_delegates[0].summary}
+            />
+          </td>
+        </tr>
+      );
+    });
+  },
+
+  _handleSummaryChange(country, event) {
+    var newSummary = event.target.value;
+    var country_assignments = this.state.country_assignments;
+    var country_delegates = country_assignments[country];
+    for (var delegate of country_delegates) {
+      delegate.summary = newSummary;
+    }
+    this.setState({country_assignments: country_assignments});
+  },
+
+  _mapAssignments() {
```

Comment:
That we need this function in the first place is pretty gross. IMO, before any more new features are built, y'all need to iron out how state is stored on the client so all of this isn't necessary.\

One thing I'm noticing is that summaries are really attached to assignments, so why don't we store them as a map from assignment ID to summary in a `summaries` state property? Then, when the user hits save or publish, we then match up the assignments with their delegates, update those fields, and send it to the server.

This way, the function can be as follows, and only needs to be called when delegates change:
```js
const summaries = {};
for (const delegate of delegates) {
  // since we're duplicating the state between the two delegates, there will be an overwrite here per delegate, but whatever.
  summaries[delegate.assignment] = delegate.summary;
}
this.setState({summaries});
```